src/SeoToolkit.Umbraco.Redirects/wwwroot/Redirects/Dialogs/import.controller.js

```diff
@@ -0,0 +1,132 @@
+        vm.import = importFile;
+        vm.validated = false;
+        vm.validationSuccess = "";
+        vm.validationFailed = "";
+        vm.importSuccess = "";
+        vm.importFailed = "";
+        vm.fileTypePropertyLabel = "Select the file type";
+        vm.fileTypePropertyDescription = "Select the type of file you want to import the redirects from";
+        vm.filePropertyLabel = "Select a file to import";
+        vm.domainPropertyLabel = "Select the domain to import for";
+        vm.domainPropertyDescription = "if nothing is selected, redirects will be imported for all sites";
+
+        var labelKeys = [
+            "redirect_fileTypePropertyLabel", "redirect_fileTypePropertyDescription",
+            "redirect_filePropertyLabel",
+            "redirect_validationSuccess", "redirect_validationFailed",
+            "redirect_importSuccess", "redirect_importFailed",
+            "redirect_domainPropertyLabel","redirect_domainPropertyDescription",
+        ];
+        localizationService.localizeMany(labelKeys).then(function (data) {
+            vm.fileTypeSelection.label = data[0];
+            vm.fileTypeSelection.description = data[1];
+            vm.fileSelection.label = data[2];
+            vm.validationSuccess = data[3];
+            vm.validationFailed = data[4];
+            vm.importSuccess = data[5];
+            vm.importFailed = data[6];
+            vm.domainPropertyLabel = data[7];
+            vm.domainPropertyDescription = data[8];
+        });
+
+        vm.fileTypes = [
+            {
+                label:"CSV",
+                extension:".csv"
+            },
+            {
+                label:"Excel",
+                extension:".xls,.xlsx"
+            },
+        ];
+
+        vm.fileTypeSelection = {
+            alias: "fileType",
+            label: vm.fileTypePropertyLabel,
+            description: vm.fileTypePropertyDescription,
+            value: 0,
+            validation: {
+                mandatory: true
+            }
+        }
+
+        vm.domainSelection = {
+            alias: "domain",
+            label: vm.domainPropertyLabel,
+            description: vm.domainPropertyDescription,
+            value: "0",
+            validation: {
+                mandatory: true
+            }
+        }
+
+        vm.fileSelection = {
+            alias: "file",
+            label: vm.filePropertyLabel,
+            value: 0,
+            validation: {
+                mandatory: true
+            }
+        }
+
+        function handleFiles(files) {
+            if (files && files.length > 0) {
+                vm.fileSelection.value = files[0];
+            }
+        }
+
+        function submit() {
+            if (formHelper.submitForm({ scope: $scope, formCtrl: $scope.createRedirectForm })) {
+                redirectsImportApiResource.validateRedirects(vm.fileTypeSelection.value.label, vm.fileSelection.value, vm.domainSelection.value).then(function (response) {
```

Comment:
I believe this should be `redirectsApiResource` as that is the property being passed in the constructor (same for in the importFile function)

Reply:
Woops, I renamed some things and caching probably prevented this from being broken :)